### PR TITLE
Switch to mutiny based reactive streams ops

### PR DESCRIPTION
**Pull request by cescoffier**
This PR provides a new extension wrapping the Mutiny based implementation of MicroProfile Reactive Streams Operators.

It also:

* switch the usages to this new extension
* bump mutiny to 0.4.3
* remove the internal reactive converter to use the one from smallrye

### CODE REVIEW

**Comment by geoand** [comment]
@cescoffier CI is failing with:

```
2020-03-19T12:34:27.0892063Z [INFO] Running io.quarkus.context.test.SimpleContextPropagationTest
2020-03-19T12:34:28.9174344Z 2020-03-19 12:34:27,172 WARN  [io.qua.dep.QuarkusAugmentor] (main) Using Java versions older than 11 to build Quarkus applications is deprecated and will be disallowed in a future release!
2020-03-19T12:34:28.9414332Z 2020-03-19 12:34:28,940 INFO  [io.agr.pool] (main) Datasource '<default>': Initial size smaller than min. Connections will be created when necessary
2020-03-19T12:34:29.3721268Z 2020-03-19 12:34:29,371 INFO  [io.quarkus] (main) Quarkus 999-SNAPSHOT started in 1.576s. Listening on: http://0.0.0.0:8081
2020-03-19T12:34:29.3724710Z 2020-03-19 12:34:29,371 INFO  [io.quarkus] (main) Profile test activated. 
2020-03-19T12:34:29.3725347Z 2020-03-19 12:34:29,371 INFO  [io.quarkus] (main) Installed features: [agroal, cdi, hibernate-orm, hibernate-orm-panache, jdbc-h2, mutiny, narayana-jta, resteasy, resteasy-mutiny, servlet, smallrye-context-propagation]
2020-03-19T12:34:31.1230065Z 2020-03-19 12:34:31,122 WARN  [org.hib.res.tra.bac.jta.int.syn.SynchronizationCallbackCoordinatorTrackingImpl] (executor-thread-2) HHH000451: Transaction afterCompletion called by a background thread; delaying afterCompletion processing until the original thread can handle it. [status=4]
2020-03-19T12:34:31.1259780Z 2020-03-19 12:34:31,123 ERROR [org.jbo.res.res.i18n] (executor-thread-2) RESTEASY002010: Failed to execute: javax.ws.rs.WebApplicationException: HTTP 409 Conflict
```

File: bom/deployment/pom.xml [hunk 1]
```diff
@@ -654,6 +654,11 @@
                 <artifactId>quarkus-mutiny-deployment</artifactId>
                 <version>${project.version}</version>
             </dependency>
+            <dependency>
+                <groupId>io.quarkus</groupId>
+                <artifactId>quarkus-mutiny-reactive-streams-operators-deployment</artifactId>
```
Comment (gsmet) [inline bom/deployment/pom.xml]: Is there a limit to the length of an artifact name? :)

File: extensions/reactive-streams-operators/mutiny-reactive-streams-operators/deployment/src/main/java/io/quarkus/mutiny/reactive/operators/deployment/MutinyReactiveStreamsOperatorsProcessor.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+import io.quarkus.deployment.builditem.FeatureBuildItem;
+import io.quarkus.deployment.builditem.nativeimage.ServiceProviderBuildItem;
+import io.smallrye.mutiny.streams.Engine;
+
+public class MutinyReactiveStreamsOperatorsProcessor {
+
+    @BuildStep
+    public void build(BuildProducer<ServiceProviderBuildItem> serviceProvider,
+            BuildProducer<FeatureBuildItem> feature) {
+        feature.produce(new FeatureBuildItem(FeatureBuildItem.MUTINY_REACTIVE_STREAMS_OPERATORS));
```
Comment (gsmet) [inline extensions/reactive-streams-operators/mutiny-reactive-streams-operators/deployment/src/main/java/io/quarkus/mutiny/reactive/operators/deployment/MutinyReactiveStreamsOperatorsProcessor.java]: I'm not sure I would add a feature if it's a purely internal extension?

**Review by gsmet — COMMENTED** [review]
I added one small question.

File: bom/deployment/pom.xml (hunk 1 above)
Comment (cescoffier) [reply bom/deployment/pom.xml]: Nope :-) Well, until it fits in the memory of the JVM.

Note that it's internal only. No one except us is going to use this.

File: extensions/reactive-streams-operators/mutiny-reactive-streams-operators/deployment/src/main/java/io/quarkus/mutiny/reactive/operators/deployment/MutinyReactiveStreamsOperatorsProcessor.java (hunk 2 above)
Comment (cescoffier) [reply extensions/reactive-streams-operators/mutiny-reactive-streams-operators/deployment/src/main/java/io/quarkus/mutiny/reactive/operators/deployment/MutinyReactiveStreamsOperatorsProcessor.java]: I agree it was more something for me to check that it was the right implementation. 
Let me remove this.

**Comment by cescoffier** [comment]
Obviously, I can't reproduce the failure.

**Comment by cescoffier** [comment]
Much better now.

The failure is not related to this PR.

**Comment by gsmet** [comment]
@cescoffier is this something we should consider for backporting?

It looks like a missing piece of the Mutiny move but I don't know how critical this is and if it's worth the risk?

**Comment by cescoffier** [comment]
Definitely no backport :-)
This is mostly a preparation for the next Reactive Messaging.

**Comment by gsmet** [comment]
I rebased it. Let's wait for CI.

**Comment by gsmet** [comment]
@cescoffier the rebase failed because of:
```
[INFO] --- maven-enforcer-plugin:3.0.0-M2:enforce (enforce) @ quarkus-mutiny-reactive-streams-operators ---
[WARNING] Rule 2: org.apache.maven.plugins.enforcer.BannedDependencies failed with message:
Found Banned Dependency: org.jboss.resteasy:resteasy-context-propagation:jar:4.5.3.Final
Use 'mvn dependency:tree' to locate the source of the banned dependencies.
```

**Comment by cescoffier** [comment]
I will rebase, but this one should not be merged before Quarkus 1.5 (so plenty of time :-))

**Comment by gsmet** [comment]
@cescoffier where are we on this one?

**Comment by cescoffier** [comment]
@gsmet targetting 1.5, I will continue it this week.

**Comment by cescoffier** [comment]
Ok, should be done now... let's see what the CI says.

**Comment by gsmet** [comment]
Force pushed to solve a conflict, let's wait for CI.

**Review by gsmet — APPROVED** [review]
Let's get this one merged!

**Pull request merged**